### PR TITLE
slepc, slepc4py

### PR DESCRIPTION
**Pull request by minrk**
follow-up now that PETSc is up

sourced from https://bitbucket.org/petsc/conda-recipes/

changes from original:

- remove MPI feature for now (only use mpich, pending MPI metapackage)
- reorder sections according to recipe lint
- remove replace-binary.py now that conda-build should do this correctly

cc @dalcinl I added you as a maintainer like on PETSc, let me know if that's okay.

### CODE REVIEW

File: recipes/slepc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{% set build = 0 %}
+{% set version = '3.7.1' %}
+{% set sha256 = '670216f263e3074b21e0623c01bc0f562fdc0bffcd7bd42dd5d8edbe73a532c2' %}
+{% set mpi = os.environ.get('MPI') or 'mpich' %}
```
Comment (scopatz) [inline recipes/slepc/meta.yaml]: Won't this always resolve to mpich on the conda-forge infrastructure?

Comment (dalcinl) [inline recipes/slepc/meta.yaml]: Yes. In the future, by using conda features, we could build the full PETSc stack with Open MPI, and provide support for both MPICH and Open MPI implementations.

Comment (scopatz) [inline recipes/slepc/meta.yaml]: I think this should either have some explanatory comments, then or leave this for the future.

Comment (minrk) [inline recipes/slepc/meta.yaml]: We can use environment variables in the [build matrix](https://github.com/conda-forge/blas-feedstock/blob/master/conda-forge.yml#L1) on feedstocks, but not on the staged-recipes repo. This is a placeholder in anticipation of that. I added a note about it, and referred to [the mpi metapackage](https://github.com/conda-forge/staged-recipes/pull/1501) before starting out supporting multiple MPIs.

Comment (scopatz) [inline recipes/slepc/meta.yaml]: Thanks @minrk!

**Comment by scopatz** [comment]
LGTM, I am going to be AFK for a bit, so if someone else whans to merge after CI is done, that would be great!

**Comment by minrk** [comment]
Thanks, @scopatz!

**Pull request merged**